src/Yasumi/Provider/AbstractProvider.php

```diff
@@ -187,22 +187,32 @@ public function removeHoliday($shortName): void
      */
     public function isWorkingDay(\DateTimeInterface $date): bool
     {
-        // First check if the given date is a holiday
-        if ($this->isHoliday($date)) {
-            return false;
-        }
+        return !$this->isHoliday($date) && !$this->isWeekendDay($date);
```

Comment:
This condition only covers holidays that fall into the weekend.

```suggestion
        return !$this->isHoliday($date) || !$this->isWeekendDay($date);
```